README.md

```diff
@@ -1,2 +1,70 @@
 # pulumi-package-publisher
 A composite action for publishing Pulumi packages
+
+_This Action is currently in preview and meant for internal use. Any functionality can and likely will change._
+
+### Purpose
+This Action automates setup and publication to the package registries of four Pulumi languages.
+
+### Setup
+
+Set the following environment variables in your Workflow and your GitHub Action secrets:
+
+```yaml
+env:
+    DOTNETVERSION: |
+    6.0.x
+    3.1.301
+    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    GOVERSION: 1.20.1
+    JAVAVERSION: "11"
+    NODEVERSION: 16.x
+    NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+    NUGET_PUBLISH_KEY: ${{ secrets.NUGET_PUBLISH_KEY }}
+    PYPI_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
+    PYTHONVERSION: "3.9"
+    SIGNING_KEY: ${{ secrets.JAVA_SIGNING_KEY }}
+    SIGNING_KEY_ID: ${{ secrets.JAVA_SIGNING_KEY_ID }}
+    SIGNING_PASSWORD: ${{ secrets.JAVA_SIGNING_PASSWORD }}
+    SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
```

Comment:
If a user doesn't use slack, can this be left blank?

Reply:
ah! yes, that will be a refactor in the future; I captured it here real quick: #5. I attempted to capture the current state of the world.